File: backend/src/main/java/com/personamarket/entity/Persona.java
package com.personamarket.entity;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "personas")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Persona {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    
    @Column(nullable = false)
    private String name;
    
    @Column(length = 2000)
    private String description;
    
    @Column(length = 5000)
    private String personality;
    
    @Column(length = 2000)
    private String tone;
    
    @Column(length = 2000)
    private String worldview;
    
    @Column(length = 5000)
    private String systemPrompt;
    
    private String avatarImage;
    
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "creator_id")
    private User creator;
    
    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private PersonaStatus status = PersonaStatus.DRAFT;
    
    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private PricingType pricingType = PricingType.FREE;
    
    private BigDecimal price;
    
    @Column(nullable = false)
    private Integer downloadCount = 0;
    
    @Column(nullable = false)
    private Double rating = 0.0;
    
    @Column(nullable = false)
    private Integer reviewCount = 0;
    
    @Column(length = 1000)
    private String tags;
    
    @Column(nullable = false)
    private Integer evolutionLevel = 1;
    
    @Column(nullable = false)
    private Integer trainingDataCount = 0;
    
    private String finetuneModelId;
    
    @OneToMany(mappedBy = "persona", cascade = CascadeType.ALL)
    @Builder.Default
    private List<Dialogue> dialogues = new ArrayList<>();
    
    @OneToMany(mappedBy = "persona", cascade = CascadeType.ALL)
    @Builder.Default
    private List<TrainingLog> trainingLogs = new ArrayList<>();
    
    @CreationTimestamp
    @Column(nullable = false, updatable = false)
    private LocalDateTime createdAt;
    
    @UpdateTimestamp
    private LocalDateTime updatedAt;
    
    public enum PersonaStatus {
        DRAFT, PUBLISHED, PRIVATE, ARCHIVED
    }
    
    public enum PricingType {
        FREE, PAID
    }
}

File: frontend/lib/models/persona.dart
class Persona {
  final int id;
  final String name;
  final String? description;
  final String? personality;
  final String? tone;
  final String? worldview;
  final String? avatarImage;
  final int creatorId;
  final String creatorUsername;
  final String status;
  final String pricingType;
  final double? price;
  final int downloadCount;
  final double rating;
  final int reviewCount;
  final String? tags;
  final int evolutionLevel;
  final int trainingDataCount;
  final DateTime createdAt;
  final DateTime updatedAt;

  Persona({
    required this.id,
    required this.name,
    this.description,
    this.personality,
    this.tone,
    this.worldview,
    this.avatarImage,
    required this.creatorId,
    required this.creatorUsername,
    required this.status,
    required this.pricingType,
    this.price,
    required this.downloadCount,
    required this.rating,
    required this.reviewCount,
    this.tags,
    required this.evolutionLevel,
    required this.trainingDataCount,
    required this.createdAt,
    required this.updatedAt,
  });

  factory Persona.fromJson(Map<String, dynamic> json) {
    return Persona(
      id: json['id'],
      name: json['name'],
      description: json['description'],
      personality: json['personality'],
      tone: json['tone'],
      worldview: json['worldview'],
      avatarImage: json['avatarImage'],
      creatorId: json['creatorId'],
      creatorUsername: json['creatorUsername'],
      status: json['status'],
      pricingType: json['pricingType'],
      price: json['price']?.toDouble(),
      downloadCount: json['downloadCount'],
      rating: json['rating']?.toDouble() ?? 0.0,
      reviewCount: json['reviewCount'],
      tags: json['tags'],
      evolutionLevel: json['evolutionLevel'],
      trainingDataCount: json['trainingDataCount'],
      createdAt: DateTime.parse(json['createdAt']),
      updatedAt: DateTime.parse(json['updatedAt']),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'description': description,
      'personality': personality,
      'tone': tone,
      'worldview': worldview,
      'avatarImage': avatarImage,
      'creatorId': creatorId,
      'creatorUsername': creatorUsername,
      'status': status,
      'pricingType': pricingType,
      'price': price,
      'downloadCount': downloadCount,
      'rating': rating,
      'reviewCount': reviewCount,
      'tags': tags,
      'evolutionLevel': evolutionLevel,
      'trainingDataCount': trainingDataCount,
      'createdAt': createdAt.toIso8601String(),
      'updatedAt': updatedAt.toIso8601String(),
    };
  }
}
